src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -67,6 +67,118 @@ public static void Resize<T>([NotNull] ref T[]? array, int newSize)
             Debug.Assert(array != null);
         }
 
+        [RequiresDynamicCode("The native code for the array might not be available at runtime.")]
+        public static unsafe Array CreateInstance(Type elementType, int length)
+        {
+            if (elementType is null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.elementType);
+            if (length < 0)
+                ThrowHelper.ThrowLengthArgumentOutOfRange_ArgumentOutOfRange_NeedNonNegNum();
+
+            RuntimeType? t = elementType.UnderlyingSystemType as RuntimeType;
+            if (t == null)
+                ThrowHelper.ThrowArgumentException(ExceptionResource.Arg_MustBeType, ExceptionArgument.elementType);
```

Comment:
```suggestion
            if (elementType.UnderlyingSystemType is not RuntimeType t)
                ThrowHelper.ThrowArgumentException(ExceptionResource.Arg_MustBeType, ExceptionArgument.elementType);
```

Reply:
This fails to compile with `error CS0165: Use of unassigned local variable 't' [C:\runtime\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.csproj]
    0`. The compiler does not see that the throw helper always throws an exception. I would have to add a dummy return statement to make it compile at which point it is kind of pointless to use the more compact syntax. Problems like this are one of the reasons why I generally dislike throwhelpers unless they are absolutely warranted for performance. One option would be to delete the throwhelpers from these methods. I do not think they are moving the needle here. I think somebody used them here just because of they can.

Reply:
@jkotas I was expecting this not to report CS0165, but ... https://github.com/dotnet/roslyn/issues/57798. I understand their rationale though.